scrnaseq_hto.Rmd

```diff
@@ -55,31 +81,94 @@ library(magrittr) # %>% operator
 # Tables: kableExtra
 # Plots: ggsci
 
-# Source plotting functions
-source("R/functions_io.R")
-source("R/functions_plotting.R")
-source("R/functions_analysis.R")
-source("R/functions_util.R")
-
 # Knitr default options
-knitr::opts_chunk$set(echo=TRUE, cache=FALSE, message=FALSE, warning=FALSE, fig.width=10)
+knitr::opts_chunk$set(echo=TRUE,                     # output code
+                      cache=FALSE,                   # do not cache results
+                      message=TRUE,                  # show messages
+                      warning=TRUE,                  # show warnings
+                      tidy=FALSE,                    # do not auto-tidy-up code
+                      fig.width=10,                  # default fig width in inches
+                      class.source='fold-hide',      # by default collapse code blocks
+                      dev=c('png', 'pdf'),           # create figures in png and pdf; the first device (png) will be used for HTML output
+                      dev.args=list(png=list(type="cairo"),  # png: use cairo - works on cluster, supports anti-aliasing (more smooth)
+                                    pdf=list(bg="white")),     # pdf: use cairo - works on cluster, supports anti-aliasing (more smooth)
+                      dpi=96                         # figure resolution
+                                     
+)
 ```
 
-```{r hto_colours}
-hto_colours = list()
-hto_colours$col_hto_global = ggsci::pal_npg()(3)
-hto_colours$col_hto_all = ggsci::pal_npg()(factorial(length(params$hto_names)) + 1)
-hto_colours$col_hto_collapsed = ggsci::pal_npg()(length(params$hto_names) + 2)
-hto_levels = c("Negative", "Doublet", params$hto_names[length(params$hto_names):1])
-names(hto_colours$col_hto_collapsed) = hto_levels
+```{r preparation_and_initial_checks}
```

Comment:
What is your motivation for all these changes and all the additional code? To make it more in line with the main script?

Reply:
Yes